### PR TITLE
typeck: minor pattern typing improvements

### PR DESCRIPTION
**Pull request by Centril**
r? @estebank

### CODE REVIEW

File: src/test/ui/rfc-2008-non-exhaustive/struct.stderr [hunk 1]
```diff
@@ -62,18 +62,33 @@ error[E0638]: `..` required with struct marked as non-exhaustive
    |
 LL |     let NormalStruct { first_field, second_field } = ns;
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+help: add `..` at the end of the field list
+   |
+LL |     let NormalStruct { first_field, second_field , .. } = ns;
+   |                                                  ^^^^^^
```
Comment (estebank) [inline src/test/ui/rfc-2008-non-exhaustive/struct.stderr]: I'd like it if we could get around pointing at the closing brace and the preceding space, but that would complicate the span gather code quite dramatically...

Comment (Centril) [reply src/test/ui/rfc-2008-non-exhaustive/struct.stderr]: Trust me, I tried several clever iterations to fix that but I got nowhere. :)

File: src/test/ui/rfc-2008-non-exhaustive/struct.stderr [hunk 2]
```diff
@@ -62,18 +62,33 @@ error[E0638]: `..` required with struct marked as non-exhaustive
    |
 LL |     let NormalStruct { first_field, second_field } = ns;
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+help: add `..` at the end of the field list
```
Comment (estebank) [inline src/test/ui/rfc-2008-non-exhaustive/struct.stderr]: Add some explanation along the lines of "add `..` at the end of the field list to ignore all other fields".

r=me

**Pull request merged**